File: app/src/main/java/com/example/techfix/data/repository/UserRepository.java
package com.example.techfix.data.repository;
import com.example.techfix.data.dao.UserDao;
import com.example.techfix.data.entity.User;
import com.example.techfix.util.PasswordUtils;
public class UserRepository {
 private final UserDao userDao;
 public UserRepository(UserDao userDao){this.userDao=userDao;}
 public User register(String name,String email,String password,String phone) throws Exception {
  if(userDao.getUserByEmail(email)!=null) throw new Exception("An account with this email already exists");
  User user=new User(name,email,PasswordUtils.hash(password),"customer",phone);
  long id=userDao.insertUser(user); user.setUserId((int)id); return user;
 }
 public User login(String email,String password) throws Exception {
  User user=userDao.getUserByEmail(email);
  if(user==null) throw new Exception("No account found with this email");
  if(!PasswordUtils.verify(password,user.getPassword())) throw new Exception("Incorrect password");
  return user;
 }
}

File: app/src/main/java/com/example/techfix/data/repository/UserRepository.kt
package com.example.techfix.data.repository

import com.example.techfix.data.dao.UserDao
import com.example.techfix.data.entity.User
import com.example.techfix.util.PasswordUtils

class UserRepository(private val userDao: UserDao) {

    suspend fun register(name: String, email: String, password: String, phone: String): Result<User> {
        val existing = userDao.getUserByEmail(email)
        if (existing != null) {
            return Result.failure(Exception("An account with this email already exists"))
        }
        val hashed = PasswordUtils.hash(password)
        val user = User(name = name, email = email, password = hashed, role = "customer", phone = phone)
        val newId = userDao.insertUser(user)
        return Result.success(user.copy(userId = newId.toInt()))
    }

    suspend fun login(email: String, password: String): Result<User> {
        val user = userDao.getUserByEmail(email)
            ?: return Result.failure(Exception("No account found with this email"))
        val passwordMatches = PasswordUtils.verify(password, user.password)
        return if (passwordMatches) {
            Result.success(user)
        } else {
            Result.failure(Exception("Incorrect password"))
        }
    }
}